packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/climatedata/ClimateDataController.java

```diff
@@ -2,70 +2,173 @@
+			if(response == null || response.getBody() == null){
+				throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Search ESGF failed.");
+			}
+
+			final List<Dataset> datasets = new ArrayList<>();
+
+			response.getBody().get("results").forEach(result -> {
+				final Dataset dataset = new Dataset();
+				dataset.setName(result.get("metadata").get("title").asText());
+				dataset.setEsgfId(result.get("metadata").get("id").asText());
+				dataset.setMetadata(result.get("metadata"));
+				datasets.add(dataset);
+			});
+
+
+			return ResponseEntity.ok(datasets);
+		} catch(final FeignException.FeignClientException e) {
+			final String error = "Unable to search ESGF";
+			final int status = e.status() >=400? e.status(): 500;
+			log.error(error, e);
+			throw new ResponseStatusException(org.springframework.http.HttpStatus.valueOf(status), error);
 
-		return ResponseEntity.ok(response.getBody());
+		} catch(final Exception e) {
+			final String error = "Unable to search ESGF";
+			log.error(error, e);
+			throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, error);
+		}
 	}
 
-	@GetMapping("/preview-esgf/{datasetId}")
+	/**
+	 * @param esgfId The id of the ESGF (in the form `CMIP6.CMIP.NCAR.CESM2.historical.r11i1p1f1.CFday.ua.gn.v20190514`)
+	 * @param variableId The variable to preview (ie `ua`)
+	 * @param timestamps Leave blank to generate for all times
+	 * @param timeIndex Leave blank to generate for all times
+	 * @return
+	 */
+	@GetMapping("/preview-esgf/{esgfId}")
 	@Secured(Roles.USER)
-	public ResponseEntity<JsonNode> previewEsgf(@PathVariable UUID datasetId,
-																							@RequestParam(value = "variable-id", required = false) final String variableId,
-																							@RequestParam(value = "timestamps", required = false) final String timestamps,
-																							@RequestParam(value = "time-index", required = false) final String timeIndex
+	@Operation(summary = "Generates a PNG for a given ESGF id")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "200", description = "URL for the generated PNG", content = @Content),
+			@ApiResponse(responseCode = "202", description = "A PNG generation request has been accepted and is being worked on", content = @Content),
+			@ApiResponse(responseCode = "500", description = "There was an issue generating the PNG", content = @Content)
+	})
+	public ResponseEntity<String> previewEsgf(@PathVariable final String esgfId,
+											  @RequestParam(value = "variable-id", required = false) final String variableId,
+											  @RequestParam(value = "timestamps", required = false) final String timestamps,
+											  @RequestParam(value = "time-index", required = false) final String timeIndex
 	) {
-		ResponseEntity<JsonNode> response = climateDataProxy.previewEsgf(datasetId.toString(), variableId, timestamps, timeIndex);
+		final ResponseEntity<String> previewResponse = climateDataService.getPreview(esgfId, variableId, timestamps, timeIndex);
```

Comment:
I'm not sure we want a Service to return `ResponseEntity`? It might be wise to let Controllers do so.
Opinion @dvince2 ?